src/index.tsx

```diff
@@ -6,14 +6,14 @@ import ReactDOM from 'react-dom'
 import { Logger } from '@/utils/logger'
 
 import { App } from './App'
-import { SENTRY_DSN } from './config/urls'
+import { ENV_DEVELOPMENT, SENTRY_DSN } from './config/urls'
 
 type Env = 'production' | 'staging' | 'development'
 
 const env = (process.env.REACT_APP_ENV?.toLowerCase() || 'development') as Env
 
 const initApp = async () => {
-  if (env === 'development') {
+  if ((env === 'development' || env === 'staging') && window.localStorage.getItem('env') === ENV_DEVELOPMENT) {
```

Comment:
Let's call this environment `mocking` instead. Given the above points the condition for enabling it would be `if (BUILD_ENV !== 'production' && TARGET_DEV_ENV === 'mocking')`